simplified-viewer-epub-readium2/src/main/java/org/librarysimplified/viewer/epub/readium2/Reader2Activity.kt

```diff
@@ -189,29 +193,25 @@ class Reader2Activity : AppCompatActivity(R.layout.reader2) {
       WebView.setWebContentsDebuggingEnabled(true)
     }
 
-    if (savedInstanceState == null) {
-      this.readerFragment =
-        this.supportFragmentManager.fragmentFactory.instantiate(
-          this.classLoader,
-          SR2ReaderFragment::class.java.name
-        )
-      this.tocFragment =
-        this.supportFragmentManager.fragmentFactory.instantiate(
-          this.classLoader,
-          SR2TOCFragment::class.java.name
-        )
+    this.readerFragment =
+      this.supportFragmentManager.fragmentFactory.instantiate(
+        this.classLoader,
+        SR2ReaderFragment::class.java.name
+      )
+    this.searchFragment =
+      this.supportFragmentManager.fragmentFactory.instantiate(
+        this.classLoader,
+        SR2SearchFragment::class.java.name
+      )
+    this.tocFragment =
+      this.supportFragmentManager.fragmentFactory.instantiate(
+        this.classLoader,
+        SR2TOCFragment::class.java.name
+      )
 
-      this.supportFragmentManager.beginTransaction()
-        .replace(R.id.reader2FragmentHost, this.readerFragment, READER_FRAGMENT_TAG)
-        .add(R.id.reader2FragmentHost, this.tocFragment, TOC_FRAGMENT_TAG)
-        .hide(this.tocFragment)
-        .commit()
-    } else {
-      this.readerFragment =
-        this.supportFragmentManager.findFragmentByTag(READER_FRAGMENT_TAG) as SR2ReaderFragment
-      this.tocFragment =
-        this.supportFragmentManager.findFragmentByTag(TOC_FRAGMENT_TAG) as SR2TOCFragment
-    }
+    this.supportFragmentManager.beginTransaction()
+      .add(R.id.reader2FragmentHost, this.readerFragment)
+      .commit()
```

Comment:
We aren't reusing fragment instances anymore?

Not that I mind, but ... does this have lifecycle repercussions?

Reply:
We are. This adds the reader fragment and then all the operations on fragments are just showing/hiding the current instances (or add one if it's not added already) so there's no recreation of fragments. In the end, when we leave the screen all the instances are destroyed alongside with the activity